webapp/src/extensionsBrowser.tsx

```diff
@@ -339,14 +332,15 @@ export const ExtensionsBrowser = (props: ExtensionsProps) => {
         }
         setPreferredExts([...repos, ...loadingCards])
 
-        fetchGithubDataAndAddAsync(toBeFetched, (exts) => setPreferredExts([...repos, ...exts]));
+        const exts = await fetchGithubDataAndAddAsync(toBeFetched);
+        setPreferredExts([...repos, ...exts])
```

Comment:
the page might be unmounted here already

Reply:
I don't think this is actually a problem in this case because this is going through the virtualapi. this will just be a no-op

Reply:
Calling setState on an unmounted component is a no-op (though it will print a warning to the console). The long-running async operations all use the virtual API (retrieving GitHub repos) which has no concept of cancellation tokens to stop operations when it no longer makes sense to continue. All we could do is check the cancellation token after the long-running virtual API has returned before calling setState, which has the same effect as just calling setState. We could add cancellation tokens to virtual API to have them return sooner, but that is outside the scope of this PR. I think adding cancellation tokens and validation functions (validating that the results are what you expect before caching them) would be useful extensions of the virtual API. Thoughts?

@pelikhan

Reply:
Leaving this be for now. We discussed creating a new hook that uses the subscribe and unsubscribe features of the virtual API to do away with the promises (thus not calling the setState functions if unsubscribed) but that work is outside the scope of this PR. Please feel free to create an issue, I'll see if I can work it in to a future PR.